scripts/start.js

```diff
@@ -81,20 +72,17 @@ const localhost = '0.0.0.0';
           return next();
         }
 
-        renderWhenReady(async ({ renderer, webpackStats }) => {
-          try {
-            const html = await renderer({
-              webpackStats,
-              route: matchingRoute.route,
-              routeName: matchingRoute.name,
-              site: getSiteForHost(req.hostname),
-              environment:
-                environments.length > 0 ? environments[0] : undefined,
-            });
-
-            res.send(html);
-          } catch (err) {
-            // Library mode does not have "devServerOnly" entry as it is a UMD
+        htmlRenderPlugin
+          .renderWhenReady({
+            route: matchingRoute.route,
+            routeName: matchingRoute.name,
+            site: getSiteForHost(req.hostname),
+            environment: environments.length > 0 ? environments[0] : undefined,
+          })
+          .then((html) => res.send(html))
+          .catch((renderError) => {
+            const webpackStats = renderError.webpackStats.toJson();
```

Comment:
I'm worried that errors aren't a happy time, and weird things happen. Can we check if webpackStats exists? It's not an expected use-case, but errors rarely work as expected :)

Reply:
Good shout 👍